code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm

```diff
@@ -134,13 +134,22 @@
 	var/damage = max((obj_integrity * 0.7) / severity, 100) //requires multiple bombs to take down
 	take_damage(damage, BRUTE, "bomb", 0)
 
-/obj/structure/destructible/clockwork/massive/celestial_gateway/proc/get_arrival_time()
+/obj/structure/destructible/clockwork/massive/celestial_gateway/proc/get_arrival_time(var/deciseconds = TRUE)
 	if(seconds_until_activation)
-		return (seconds_until_activation*10)
+		if(deciseconds)
+			return (seconds_until_activation * 10)
+		else
+			return seconds_until_activation
 	if(grace_period)
-		return (grace_period*10)
+		if(deciseconds)
```

Comment:
this code could be compressed with use of `.`, something like this, with it being the entire block
````
/obj/structure/destructible/clockwork/massive/celestial_gateway/proc/get_arrival_time(var/deciseconds = TRUE)
 	if(seconds_until_activation)
		. = seconds_until_activation
	else if(grace_period)
		. = grace_period
	else if(GATEWAY_RATVAR_ARRIVAL - progress_in_seconds > 0)
		. = round(max((GATEWAY_RATVAR_ARRIVAL - progress_in_seconds) / (GATEWAY_SUMMON_RATE), 0), 1)
	if(deciseconds)
		. *= 10
````